src/SkeletalTrapezoidationEdge.h

```diff
@@ -39,11 +70,53 @@ class SkeletalTrapezoidationEdge
     {
         return is_marked >= 0;
     }
+
+    bool hasTransitions(bool ignore_empty = false) const
+    {
+        return transitions_.use_count() > 0 && (ignore_empty || ! transitions_.lock()->empty());
+    }
+    void initTransitions(std::shared_ptr<std::list<TransitionMiddle>> storage)
+    {
+        transitions_ = storage;
+    }
+    std::shared_ptr<std::list<TransitionMiddle>> transitions()
+    {
+        return transitions_.lock();
+    }
+
+    bool hasTransitionEnds(bool ignore_empty = false) const
+    {
+        return transition_ends_.use_count() > 0 && (ignore_empty || ! transition_ends_.lock()->empty());
+    }
+    void initTransitionEnds(std::shared_ptr<std::list<TransitionEnd>> storage)
+    {
+        transition_ends_ = storage;
+    }
+    std::shared_ptr<std::list<TransitionEnd>> transition_ends()
```

Comment:
It's technically a function, so it should be transitionEnds (or getTransitionEnds).